src/core/lib/slice/slice_buffer.cc

```diff
@@ -36,16 +36,21 @@
  * memmove/malloc/realloc) - only if we were up against the full capacity of the
  * slice buffer. If do_embiggen is inlined, the compiler clobbers multiple
  * registers pointlessly in the common case. */
+template <bool grow_past_minimum = false>
 static void GPR_ATTRIBUTE_NOINLINE do_embiggen(grpc_slice_buffer* sb,
                                                const size_t slice_count,
-                                               const size_t slice_offset) {
+                                               const size_t slice_offset,
+                                               const size_t minimum_size = 0) {
```

Comment:
Again, let me suggest avoiding the use of default parameters and arguments for internal functions.